### PR TITLE
Make port and host configurable in wpds, make library more dynamic

### PR DESCRIPTION
**Pull request by illepic**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -101,15 +100,14 @@ gulp.task('webpack:watch:scss-to-json', (cb) => {
...
-const localhost = 'http://localhost:8080';
 // Import webpack config for PL
 const wpconfig = require('./webpack.pl.config');
 // Webpack Dev Server config used for local development.
 // See all available config options:
 // https://webpack.js.org/configuration/dev-server/#devserver
 const serverconfig = {
-  publicPath: url.resolve(localhost, wpconfig.output.publicPath), // ie http://localhost:8080/temp
+  host: '0.0.0.0',
+  port: '8080',
```
Comment (grayside) [inline gulpfile.js]: This can be overridden from webpack.pl.config?

Comment (illepic) [reply gulpfile.js]: There is zero webpack dev server config in our webpack configs. It's either handled at our [webpack-dev-server.js](https://github.com/phase2/pattern-lab-starter/pull/133/files#diff-1763473eec6d663f46bbaecfdf53009e) service or within this gulpfile.js task.

File: tools/tasks/webpack-dev-server.js [hunk 2]
```diff
@@ -12,29 +13,39 @@ let wpds = null; // Hold a reference to Webpack Dev Server when it is created
...
  *  - Added entry points
  */
-module.exports = function startWebpackDevServer(webpackConfig, localhost, devServerConfig) {
+module.exports = function startWebpackDevServer(webpackConfig, devServerConfig) {
   return (cb) => {
     const localWebpackConfig = webpackConfig;
+    const {
+      entry: { 'pattern-lab': plEntry }, // ./source/pattern-lab.js
+      output: { publicPath }, // /temp/
+    } = webpackConfig;
```
Comment (thagler) [inline tools/tasks/webpack-dev-server.js]: Is this the magic that handles the path issues when served from within a container?

File: gulpfile.js (hunk 1 above)
Comment (grayside) [reply gulpfile.js]: Well, the thing is that binding to 0.0.0.0:80 is really aggressive outside the context of a container.

**Pull request merged**